src/client/dfs/cont.c

```diff
@@ -83,7 +84,7 @@ dfs_cont_create(daos_handle_t poh, uuid_t *cuuid, dfs_attr_t *attr, daos_handle_
 			dattr.da_chunk_size = DFS_DEFAULT_CHUNK_SIZE;
 
 		if (attr->da_hints[0] != 0) {
-			strncpy(dattr.da_hints, attr->da_hints, DAOS_CONT_HINT_MAX_LEN);
+			strncpy(dattr.da_hints, attr->da_hints, DAOS_CONT_HINT_MAX_LEN - 1);
 			dattr.da_hints[DAOS_CONT_HINT_MAX_LEN - 1] = '\0';
```

Comment:
The implementation is actually perfectly fine. Are you see any issues with the original code?

Reply:
Yes, please take a look at #15105

Reply:
I do think the existing code has a bug. If `attr->da_hints` is exactly `DAOS_CONT_HINT_MAX_LEN` long, then strncpy copies the entire buffer as expected. But then `dattr.da_hints[DAOS_CONT_HINT_MAX_LEN - 1] = '\0';` overwrites the last character.
E.g.
```
strncpy(dattr.da_hints, "1234", 4);
dattr.da_hints[4 - 1] = '\0';
```
Yields `dattr.da_hints = "123\0"`

Reply:
yes, finally the string will be somehow OK, but it will be not the same string. The problem of strncpy misusage has been discussed in #15105

Reply:
But then again I guess it doesn't matter because `attr->da_hints` could be longer than `DAOS_CONT_HINT_MAX_LEN` and not copied entirely anyway

Reply:
Reading up on this more, it's not actually a misuse of strncpy, but just simply that `-Wstringop-truncation` warns that you might lose data from the source string.

Reply:
Also, doing some local testing, the compiler complains either way. So this change doesn't solve anything

With
```
char my_string[4];
strncpy(my_string, "1234", 4);
my_string[4-1] = '\0';
```
It complains with
```
warning: ‘strncpy’ output truncated before terminating nul copying 4 bytes from a string of the same length [-Wstringop-truncation]
     strncpy(my_string, "1234", 4);
```

And with
```
char my_string[4];
strncpy(my_string, "1234", 4-1);
my_string[4-1] = '\0';
```
It complains with
```
warning: ‘strncpy’ output truncated copying 3 bytes from a string of length 4 [-Wstringop-truncation]
     strncpy(my_string, "1234", 4-1);
```

Reply:
I see. thanks for the explanation